pandas/core/internals.py

```diff
@@ -2753,6 +2761,17 @@ def _astype(self, dtype, copy=False, raise_on_error=True, values=None,
         return self.make_block_same_class(values=values,
                                           placement=self.mgr_locs)
 
+    def _can_hold_element(self, element):
+        return np.can_cast(np.asarray(element).dtype, self.sp_values.dtype)
+
+    def _try_coerce_result(self, result):
+        if (isinstance(result, np.ndarray) and
```

Comment:
A failing test where a SparseDataFrame is sliced horizontally requires this check. In that test, the items of a dtype=object array are empty lists. Were they ndarray objects, this method would coerce them into SparseArrays (i.e. no longer pure ndarrays).